11-payment-encoding.md

```diff
@@ -140,6 +140,8 @@ Currently defined tagged fields are:
 * `p` (1): `data_length` 52. 256-bit SHA256 payment_hash. Preimage of this provides proof of payment.
 * `s` (16): `data_length` 52. This 256-bit secret prevents forwarding nodes from probing the payment recipient.
 * `d` (13): `data_length` variable. Short description of purpose of payment (UTF-8), e.g. '1 cup of coffee' or 'ナンセンス 1杯'
+* `m` (14): `data_length` variable. Additional metadata to attach to the
+  payment. Note that implementations may apply limits to the size of this field because the size of the hop payload itself is limited as well.
```

Comment:
Can decode it alright, but when I encode myself I get `lnbc10m1pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqdp9wpshjmt9de6zqmt9w3skgct5vysxjmnnd9jx2wq8q8a04uqnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66sp5zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zygs9q2gqqqqqqsgq7h4ea65u4zwf4j6jjwru7wqcvdtes8z54pphp08as630aljqvlxpyxrde8ddfyy8fggcctlzkqwvhggd6lkg00406urur6yc2p8djwgp6tyhkv`.

One difference seems to be the order of the fields. Is there a requirement for order?

Reply:
Btw, do you have an easy way to get that bolt11 string decomposition that the other test cases have?

Reply:
Okay, there are some problems here. `m` isn't 14. Will get back with a proper vector.

Reply:
Okay, I have added a test vector with decomposition. Tag `m` is now number 27 as is should be.

Reply:
LGTM, I added your test vector to https://github.com/ACINQ/eclair/pull/2063 and it works fine.
Good catch on the `m (14)` instead of `m (27)`, I somehow missed that.